File: leet_code_solutions.py/28.py
# Runtime: 32 ms, faster than 77.54% of Python3 online submissions for Implement strStr().
# Memory Usage: 14.4 MB, less than 81.36% of Python3 online submissions for Implement strStr().

class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if not needle:
            return 0
    
        return haystack.index(needle) if needle in haystack else -1

File: leet_code_solutions.js/28.js
// Solved
// Runtime: 72 ms, faster than 88.39% of JavaScript online submissions for Implement strStr().
// Memory Usage: 40.1 MB, less than 44.22% of JavaScript online submissions for Implement strStr().

/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
  if (needle === "") return 0;

  if (haystack.includes(needle)) {
    return haystack.indexOf(needle);
  } else {
    return -1;
  }
};
